timemachine/cpp/src/nonbonded_all_pairs.cu

```diff
@@ -243,27 +264,23 @@ void NonbondedAllPairs<RealType>::execute_device(
 
     const int tpb = warp_size;
 
-    // (ytz) see if we need to rebuild the neighborlist.
-    k_check_rebuild_coords_and_box_gather<RealType><<<ceil_divide(K_, tpb), tpb, 0, stream>>>(
-        K_, d_atom_idxs_, d_x, d_nblist_x_, d_box, d_nblist_box_, nblist_padding_, d_rebuild_nblist_);
-    gpuErrchk(cudaPeekAtLastError());
+    if (this->needs_sort()) {
+        // Will require a nblist rebuild by setting p_rebuild_nblist to nonzero
+        this->sort(d_x, d_box, stream);
+    } else {
```

Comment:
It seems a bit confusing that we skip the neighborlist rebuild check on iterations where we sort. Is this intentional, or was it just more convenient to implement this way? (If the latter, could also imagine something factoring out the code to rebuild the neighborlist into a method and calling directly from `sort`)

Reply:
This is intentional. 

We don't need to run the other kernels to check if we need to rebuild because by sorting we will have to a rebuild. A minor optimization on the steps that we know we will have to rebuild regardless of the coordinates.